src/main/java/seedu/utility/Ui.java

```diff
@@ -102,18 +102,48 @@ public void listExpense(ArrayList<Expense> expenses) {
      */    
     public void listIncome(ArrayList<Income> incomes) {
         printLine();
+        if (incomes.isEmpty()) {
+            printEmptyIncomeListMessage();
+        } else {
+            printIncomeList(incomes);
+        }
+        printLine();
+    }
+
+    private void printEmptyIncomeListMessage() {
+        System.out.println(EMPTY_INCOME_MESSAGE);
+    }
+
+    private void printEmptyExpenseListMessage() {
+        System.out.println(EMPTY_EXPENSE_MESSAGE);
+    }
+    
+    private void printIncomeList(ArrayList<Income> incomes) {
         System.out.println(LISTING_INCOME_MESSAGE);
         printLine();
-        int i = 1;
 
-        for (Income income:incomes) {
+        int i = 1;
+        for (Income income : incomes) {
             System.out.print(i + ": ");
             System.out.println(income);
             i++;
         }
+    }
+
+    private void printExpenseList(ArrayList<Expense> expenses) {
+        System.out.println(LISTING_EXPENSE_MESSAGE);
         printLine();
+        
+        int i = 1;
+        for (Expense expense : expenses) {
+            System.out.print(i + ": ");
```

Comment:
Perhaps you might want to refactor line 139 and 140 into a single method for method readability as well